content/en/docs/how-tos/artifacts.md

```diff
@@ -0,0 +1,340 @@
+│   │   ├── …
+│   │   └── src.log
+│   ├── build-resources
+│   │   ├── builds.json
+│   │   ├── events.json
+│   │   ├── imagestreams.json
+│   │   ├── pods.json
+│   │   └── templateinstances.json
+│   ├── test
+│   │   ├── artifacts
+│   │   │   ├── …
+│   │   │   └── junit.xml
+│   │   ├── build-log.txt
+│   │   ├── finished.json
+│   │   └── sidecar-logs.json
+│   ├── ci-operator-step-graph.json
+│   ├── ci-operator.log
+│   ├── junit_operator.xml
+│   └── metadata.json
+├── build-log.txt
+├── clone-log.txt
+├── clone-records.json
+├── finished.json
+├── podinfo.json
+├── prowjob.json
+├── sidecar-logs.json
+└── started.json
+{{< / highlight >}}
+
+- `artifacts/`: the `ci-operator` artifacts, described [below]({{< ref
+  "#ci-operator-artifacts" >}}).
+
+- `build-log.txt`: the build log as a text file, which also appears in the
+  "build log" section of the main page.
+
+- `clone-log.txt`: if the Prow job itself clones the repository under test, this
+  file will contain the main `clonerefs` output.  It corresponds to the
+  `clonerefs` section at the bottom of the main page.
+
+- `clone-records.json`: same as the previous item, but the raw output in JSON
+  format.
+
+- `finished.json`: the file uploaded by the either the `initupload` or `sidecar`
+  containers when the test finishes.
+
+- `podinfo.json`: a dump of the `Pod` and `Event` objects associated with the
+  test, also found in the "Pod" and "Events" sections of the main page.
+
+- `prowjob.json`: a dump of the `ProwJob` object, also found in the "Prow Job
+  YAML" link in the main page.
+
+- `sidecar-logs.json`: output of the `sidecar` container, which is responsible
+  for uploading the files in this directory.  It corresponds to the `sidecar`
+  section at the bottom of the main page.
+
+- `started.json`: file uploaded by the `initupload` _init_ container when the
+  test starts.
+
+### `ci-operator` Artifacts
+
+A test executed by `ci-operator` will also have a directory called `artifacts`
+among the files described in the previous section.  This is the first directory
+in the previous listing and has the following contents:
+
+{{< highlight bash >}}
+.
+├── build-logs
+│   ├── …
+│   └── src.log
+├── build-resources
+│   ├── builds.json
+│   ├── events.json
+│   ├── imagestreams.json
+│   ├── pods.json
+│   └── templateinstances.json
+├── test
+│   ├── artifacts
+│   │   ├── …
+│   │   └── junit.xml
+│   ├── build-log.txt
+│   ├── finished.json
+│   └── sidecar-logs.json
+├── ci-operator-step-graph.json
+├── ci-operator.log
+├── junit_operator.xml
+└── metadata.json
+{{< / highlight >}}
+
+The files immediately under the directory are not commonly useful to test
+authors, they are mostly meant for test platform engineers:
+
+- `ci-operator-step-graph.json`: this is a dump of every Kubernetes/OpenShift
+  object created by each step (n.b.: "step" here means `ci-operator` step, not a
+  step of a multi-stage test &mdash; the latter being a type of the former).
```

Comment:
Correct me if I am wrong: a step is the primitive unit of execution inside the `ci-operator` execution graph. A multi-stage test is a step (or maybe "a set of steps") of that graph. It means tests for `ci-operator` are nothing special but just regulars steps as the others, like: cloning repositories, building images, promoting images, etc.

Reply:
That's correct.  We document all (`ci-operator`) steps [here](https://docs.ci.openshift.org/docs/architecture/ci-operator-internals/steps/).  The ones you mentioned are:

- [tests (multiple types)](https://docs.ci.openshift.org/docs/architecture/ci-operator-internals/steps/#test-steps)
- [source clone](https://docs.ci.openshift.org/docs/architecture/ci-operator-internals/steps/#sourcestep)
- [image builds](https://docs.ci.openshift.org/docs/architecture/ci-operator-internals/steps/#projectdirectoryimagebuildstep)
- [promotion](https://docs.ci.openshift.org/docs/architecture/ci-operator-internals/steps/#promotionstep)

The distinction made in this paragraph is to clarify ambiguous terminology since each individual test in a multi-stage test (itself a `ci-operator` [step](https://docs.ci.openshift.org/docs/architecture/ci-operator-internals/steps/#multistageteststep)) is also called a ["step"](https://docs.ci.openshift.org/docs/architecture/step-registry/#step).

So `ci-operator` steps constitute the main execution graph, and the steps of a multi-stage test constitute a sub-graph (conceptually, that's not how it's implemented) of that individual test step.